src/modules/verifiable-presentation/services/vc-match.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@nestjs/common';
+import { PEX } from '@sphereon/pex';
+import { Logger } from '../../logger/logger.service';
+import { MessageService } from '../../decentralized-web-node/services/message.service';
+import { MatchResultsDto } from '../dtos/match-results.dto';
+import { PresentationDefinitionDto } from '../dtos/presentation-definition.dto';
+
+@Injectable()
+export class VCMatchService {
+  constructor(
+    private readonly logger: Logger,
+    private readonly messageService: MessageService
+  ) {
+    this.logger.setContext(VCMatchService.name);
+  }
+
+  async matchCredentials(
+    presentationDefinition: PresentationDefinitionDto,
+    holder: string
+  ): Promise<MatchResultsDto> {
+    presentationDefinition;
+    holder;
+    const pex: PEX = new PEX();
+    pex;
+    this.messageService;
+    // const vcStorage = await this.vcStorageService.findAll(holder);
+    // const credentials = vcStorage.map((vc) => vc.vc);
+
+    // return pex.selectFrom(presentationDefinition, credentials, [holder]);
+    return [] as any;
```

Comment:
I guess this still needs to be wired up to the decentralized-web-node module. But the interface looks good!

Reply:
First I finish `decentralized-web-node` module then I will switch to this module. I was thinking about writing query messages and internally send it to `HandlerService`, what's your opinion on this approach?

Reply:
>  I was thinking about writing query messages and internally send it to HandlerService, what's your opinion on this approach?

This sounds good the me 👍